src/services/inspection.ts

```diff
@@ -0,0 +1,13 @@
+import { Artifact } from '@/models'
+import { emitter } from '@/objects/events'
+import { getSelected, selectNode } from '@/objects/selection'
+
+export function inspectArtifact(artifact: Artifact): void {
+  const selectedNode = getSelected()
+
+  if (selectedNode && !selectedNode.artifacts?.includes(artifact)) {
```

Comment:
When a node is selected, clicking an un-highlighted node to select/highlight it works pretty intuitively. It was weird to have a path highlighted, then click an artifact on an un-highlighted node and have the artifact drawer appear, but not affect the highlighting at all. I opted here to simply deselect at that point.